ex2_system/source/beacon/beacon_task.c

```diff
@@ -13,96 +13,43 @@
  */
 /**
  * @file beacon_task.c
- * @author Andrew R. Rooney
+ * @author Andrew R. Rooney, Grace Yi
  * @date Mar. 6, 2021
  */
 
-#include <beacon_task.h>
-#include <FreeRTOS.h>
-#include <os_task.h>
-
-#include "HL_gio.h"
-#include "HL_het.h"
-#include "eps.h"
-#include "uhf.h"
+#include "beacon_task.h"
 
 #define SCW_BCN_FLAG 5
 #define SCW_BCN_ON 1
+#define BEACON_PACKET_LENGTH 97
 
-typedef struct __attribute__((packed)) {
-    uint32_t time;
-    uint8_t packet_number;
-    uint8_t eps_mode;
-    uint16_t battery_voltage;
-    uint16_t battery_input_current;
-    uint16_t current_channels[10];
-    uint16_t output_states;
-    uint8_t output_faults[10];
-    uint16_t EPS_boot_count;
-    uint8_t eps_last_reset_reason;
-    uint16_t gs_wdt;
-    uint8_t obc_wdt;
-    uint16_t gs_wdt_expr;
-    uint16_t obc_wdt_expr;
-    int8_t temps[18];
-} beacon_common_t;
-typedef struct __attribute__((packed)) {
-    beacon_common_t common;
-    int8_t angular_rate;
-    int8_t adcs_control_mode;
-    uint16_t uhf_uptime;
-    uint8_t payload_software_version;
-    uint16_t obc_boot_count;
-    uint8_t obc_last_reset_reason;
-    uint8_t obc_mode;
-    uint16_t obc_uptime;
-    uint8_t solar_panel_current;
-    uint16_t mcu_core_current;
-    uint8_t obc_software_version;
-    uint16_t commands_received;
-    uint16_t fec_recovered_packets;
-    uint8_t logged_items_count;
-    uint32_t log1_timestamp;
-    uint8_t log1_code;
-} beacon_packet_1_t;
-typedef struct __attribute__((packed)) {
-    beacon_common_t common;
-    uint32_t log2_timestamp;
-    uint8_t log2_code;
-    uint32_t log3_timestamp;
-    uint8_t log3_code;
-    uint32_t log4_timestamp;
-    uint8_t log4_code;
-    uint32_t log5_timestamp;
-    uint8_t log5_code;
-    uint32_t log6_timestamp;
-    uint8_t log6_code;
-} beacon_packet_2_t;
-
-static void *beacon_daemon(void *pvParameters);
-SAT_returnState start_beacon_daemon(void);
+static void *beacon_daemon(All_systems_housekeeping* all_hk_data);
+SAT_returnState start_beacon_daemon();
 static TickType_t beacon_delay = pdMS_TO_TICKS(1000); //converts 1000 ms to number of ticks
 /**
  * Construct and send out the system beacon at the required frequency.
  *
  * @param pvParameters
  *    task parameters (not used)
  */
-static void *beacon_daemon(void *pvParameters) {
+static void *beacon_daemon(All_systems_housekeeping* all_hk_data) {
+    TickType_t beacon_update_delay = pdMS_TO_TICKS(1000);
+    uint32_t seconds_delay = 30;
     for (;;) {
         int8_t uhf_status = -1;
         /* Constructing the system beacon content */
         // Refer to table 3 of MOP
         UHF_configStruct beacon_msg;
-        All_systems_housekeeping all_hk_data;
+        //All_systems_housekeeping all_hk_data;
 
         //Testing purposes only, passing 1s to the UHF
         beacon_packet_1_t beacon_packet_one = {
             //define each element as 1 for testing purposes
             1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1};

```

Comment:
Is this test still needed?

Reply:
yes, we never got to fully test the beacon because UHF code could not handle zeros (it used strlen which is NULL terminated, should switch it to sizeof).